src/EditorFeatures/Test2/IntelliSense/CSharpCompletionCommandHandlerTests.vb

```diff
@@ -4722,7 +4722,7 @@ class C
 
                 state.SendTypeChars("M(M(M(M(")
                 Await state.AssertNoCompletionSession()
-                Assert.Equal("        Ma(m:(Ma(m:(", state.GetLineTextFromCaretPosition())
+                Assert.Equal("        Ma(m(Ma(m(", state.GetLineTextFromCaretPosition())
```

Comment:
isn't this undoing intentional behavior we were testing for?

Reply:
> isn't this undoing intentional behavior we were testing for?
@CyrusNajmabadi , @sharwell  this is exactly the question. As I remember, this specific scenario was introduced during implementation of the async completion. So, it is not a scenario supported by muscle memory of thousand developers.

We attempted to make a special casing for this and similar scenarios: we started adding colon always except commit on tab. This rise the bug we are currently fixing.

The first question was: do we want to add colon in case of nonexistent identifiers. The answer was - No.

Then, I left the special case for `(`. @sharwell suggested not to special case any scenario. I do not have a definite preference for this special case:
1. It is not legacy enough to be supported.
2. My personal preference is to add colon in all cases if we do not have existing identifier. I do not have the opposite muscle memory and adding colon looks more logical for me.

So, please let me know your preference for the special casing.

Reply:
@sharwell @CyrusNajmabadi any comments?

Reply:
this is a tough space.  we have the potential to screwup muscle memory for users no matter waht.  I think i would personally just say go with sharwell's approach.  Perhaps a design meeting to just to make sure the team is ok with it.  But i like the idea that the information is in the completion list *just* as documentation, but the user always need to type it. That very much fits other things like `List<>`.  You never get `<>` inserted.  It's just decoration to help you know it's a generic item.

I would *even* go as far as saying: the item in the completion list should just be `paramName` and not `paramName:`.  Then, we don't even confuse users who say "but why did you not insert the `:`???"

Reply:
Thank you, @CyrusNajmabadi !

Since our strategy is too keep users' muscle memory, let us follow @sharwell 's approach, which is currently implemented. @sharwell , could you please review?

Reply:
> I would even go as far as saying: the item in the completion list should just be paramName and not paramName:.

We do the same thing for generics. It might be confusing at the start but not many people ask about it so I'm guessing it's pretty easy to figure out.